processors/base/helpers/__init__.py

```diff
@@ -271,6 +270,54 @@ def find_trial_by_identifiers(conn, identifiers, ignore_record_id=None):
     return trial
 
 
+def find_trial(conn, trial, source_id, record_id=None):
+    """Find if the trial already exists. Firstly, via identifiers. Secondly, via public title
+    Args:
+        conn (dict): connection dict
+        trial (dict): trial that may be found
+        source_id (text): trial's source
+        record_id (str): record id from warehouse
+    Returns:
+        dict: trial
+    """
+
+    def write_log(trial_id, record_id, method):
+        log = conn['database']['trial_deduplication_logs'].find_one(
+            trial_id=trial_id, record_id=record_id, method=method)
+
+        if not log:
+            repo = git.Repo(search_parent_directories=True)
+            commit_sha = repo.head.object.hexsha
+
+            data = {
+                'trial_id': trial_id,
+                'record_id': record_id,
+                'method': method,
+                'commit': commit_sha
+            }
+            conn['database']['trial_deduplication_logs'].insert(data)
+
+    trial_found = find_trial_by_identifiers(conn, trial['identifiers'],
+                                            ignore_record_id=record_id)
+
+    if not trial_found:
+        trial_found = find_trial_by_identifiers(conn, trial['identifiers'])
+
+    if trial_found:
+        logger.debug('Trial-id %s was matched via identifiers with register-id %s',
+                     trial_found['id'], record_id)
+        write_log(trial_found['id'], record_id, 'identifier')
+    elif source_id and trial.get('trial_public_title'):
+        trial_temp = conn['database']['trials'].find_one(
+                            public_title=trial.get('trial_public_title'))
+        if trial_temp and trial_temp.get('source_id') != source_id:
```

Comment:
Ok, now it's clear. 
We already made the commit that fixes the first case you just explained.